pkg/podman/listener.go

```diff
@@ -0,0 +1,46 @@
+package podman
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	podmanArgsBase        = "--log-level=debug system service -t 0"
+	podmanExec            = "/usr/bin/podman"
+	podmanListenerLogFile = "podman-system-service.log"
+)
+
+// creates a listening service that answers API calls for Podman (https://docs.podman.io/en/v4.8.2/markdown/podman-system-service.1.html)
+// only way to start the service from within a container - https://github.com/containers/podman/tree/v4.8.2/pkg/bindings#starting-the-service-manually
+func setupAPIListener(out string) error {
+	cmd, logfile, err := preparePodmanCommand(out)
```

Comment:
The problem with this is that it's much harder to test the arguments of the command. It's more flexible to create the log file and only then run `preparePodmanCommand` with an `io.Writer` argument.

Reply:
Agreed, I was using the [xorriso](https://github.com/suse-edge/edge-image-builder/blob/main/pkg/build/iso.go#L48) prepare command function for inspiration, but I definitely see the value in your suggestion.